src/uwtools/config/tools.py

```diff
@@ -38,6 +38,24 @@ def compare_configs(
     return cfg_a.compare_config(cfg_b.data)
 
 
+def config_check_depths(input_obj: Config, target_format: Optional[str]) -> None:
+    """
+    Check that the depth of the input config does not exceed the target format's max.
+
+    :param input_obj: The input config.
+    :param target_format: The target config.:
+    :raises: UWConfigError on excessive input-config depth.
+    """
+    cfgclass = format_to_config(str(target_format))
+    if cfgclass.DEPTH and input_obj.depth != cfgclass.DEPTH:
```

Comment:
I understand this was the original code, and it's just been moved into the appropriate place when it was renamed. One interesting thing that our conversations have brought up, though, is that we don't require a values file (now called supplemental files) to be the _same_ depth, but we do require that it doesn't exceed the target depth (as mentioned in the doc string above).

How should we approach a check where it can handle both cases?

Reply:
Thanks  - I wanted to make sure my work was up before I left for a break. Hmm, I reflexively wanted to make it `!<=` but that's not quite right. It will have to know 'why' it was called, either separating it into 2+ functions or by passing it a flag. I'm inclined to give it a flag while also moving the format assert versions back into here

Reply:
Error logging now added